### PR TITLE
[TF FE] Add layer tests for If operation

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Add layer tests for If operation

**Tickets:** 92783, 92832

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_If.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    def _prepare_input(self, inputs_info):
+        assert 'cond' in inputs_info, "Test error: inputs_info must contain `cond`"
+        assert 'ind' in inputs_info, "Test error: inputs_info must contain `ind`"
+        assert 'y' in inputs_info, "Test error: inputs_info must contain `y`"
+        cond_shape = inputs_info['cond']
+        ind_shape = inputs_info['ind']
+        y_shape = inputs_info['y']
+        inputs_data = {}
+        inputs_data['cond'] = np.random.randint(0, 2, cond_shape).astype(bool)
+        inputs_data['ind'] = np.random.randint(1, 10, ind_shape).astype(np.int32)
```
Comment (itikhono) [inline tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: is the difference between these 2 tests only in the input types?
why can't we parameterize them?

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: not only types, but also we have difference in bodies.

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: also, in the second case there is a mix of types in inputs and outputs

Comment (itikhono) [reply tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: the test names are confusing in this case "TestIfInt" and "TestIfFloat". Please describe the difference in the comment or select other names.

Comment (gkrivor) [reply tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: Overall - agree. Also I don't understand where is difference between "x" and "ind". I also see some difference between expressions, but not sure I understand are they meaningful or not.
I suggest to carefully review this part, but don't think it is merge blocker

**Review by gkrivor — APPROVED** [review]
Non-blocking comment from my side

File: tests/layer_tests/tensorflow_tests/test_tf_If.py (hunk 1 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_If.py:80]: @itikhono, well, let me fix this in the separate PR. I think this is too minor.
Will they be fine to call as `TestIfWithFloatInputs` and `TestIfWithMixedTypes`?

**Review by itikhono — APPROVED** [review]

**Pull request merged**